### PR TITLE
Avoid creating an illegal pointer

### PR DESCRIPTION
**Pull request by kroeckx**
Found by tis-interpreter

### CODE REVIEW

File: crypto/asn1/tasn_fre.c [hunk 1]
```diff
@@ -102,9 +102,11 @@ static void asn1_item_embed_free(ASN1_VALUE **pval, const ASN1_ITEM *it,
...
          * defines. So free up in reverse order.
          */
-        tt = it->templates + it->tcount - 1;
-        for (i = 0; i < it->tcount; tt--, i++) {
+        tt = it->templates + it->tcount;
+        for (i = 0; i < it->tcount; i++) {
             ASN1_VALUE **pseqval;
+
+            tt--;
             seqtt = asn1_do_adb(pval, tt, 0);
```
Comment (richsalz) [inline crypto/asn1/tasn_fre.c]: could merge by using ```--tt``` as the param

**Comment by richsalz** [comment]
+1

File: crypto/asn1/tasn_fre.c (hunk 1 above)
Comment (kroeckx) [inline crypto/asn1/tasn_fre.c]: I really prefer writing this on multiple lines, and I guess you don't.

Comment (richsalz) [inline crypto/asn1/tasn_fre.c]: just pointing it out, that's all.  still +1 from me.

**Pull request closed without merging**